flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannel.java

```diff
@@ -340,8 +340,10 @@ void releaseAllResources() throws IOException {
     void announceBufferSize(int newBufferSize) {
         checkState(!isReleased, "Channel released.");
 
-        ResultSubpartitionView subpartitionView = checkNotNull(this.subpartitionView);
-        subpartitionView.notifyNewBufferSize(newBufferSize);
+        ResultSubpartitionView view = this.subpartitionView;
+        if (view != null) {
+            view.notifyNewBufferSize(newBufferSize);
```

Comment:
Not sure, if a better solution, just throwing the idea out. Would it make sense to do the same thing as in `getNextBuffer` and call `checkAndWaitForSubpartitionView`?

BTW, is the long term fix aligned with the comment there? Do you plan to fix the `getNextBuffer` as well in FLINK-24328?

Reply:
It is not a same problem as in `getNextBuffer`.  the data race inside of `getNextBuffer` is possible only if `requestSubpartition` was called already. in the case of `announceBufferSize`, it can be called at any moment when the first `requestSubpartition` was failed for the first time and was rescheduled. So `checkAndWaitForSubpartitionView` doesn't help here.
It is definitely not the best solution but it is better than failing at the start and it is not a big problem if we skip the first buffer size announcement. But of course, we should fix this.

> is the long term fix aligned with the comment there? Do you plan to fix the getNextBuffer as well in FLINK-24328?

I didn't know about this problem so initially, it wasn't a target to fix it. But if the solution will be similar we can think about that.

Reply:
Makes sense, thanks for the explanation!

I don't know what was your plan for the long term fix, but I'd suggest considering if we can fix both places at the same time.